sonic_y_cable/credo/y_cable_credo.py

```diff
@@ -1463,14 +1463,21 @@ def get_firmware_version(self, target):
         if self.platform_chassis is not None:
             with self.rlock.acquire_timeout(RLocker.ACQUIRE_LOCK_TIMEOUT) as lock_status:
                 if lock_status:
+                    curr_offset = YCable.OFFSET_DETERMINE_CABLE_READ_SIDE
+                    read_side = self.platform_chassis.get_sfp(self.port).read_eeprom(curr_offset, 1)
+
                     vsc_req_form = [None] * (YCable.VSC_CMD_ATTRIBUTE_LENGTH)
                     vsc_req_form[YCable.VSC_BYTE_OPCODE] = YCable.VSC_OPCODE_FWUPD
                     vsc_req_form[YCable.VSC_BYTE_OPTION] = YCable.FWUPD_OPTION_GET_INFO
                     status = self.send_vsc(vsc_req_form)
 
                     if status != YCable.MCU_EC_NO_ERROR:
-                        self.log_error('Get firmware version error (error code:0x%04X)' % (status))
-                        return None
+                        ''' should at least return local side fw version if nic is offline'''
+                        if status == 0x1A and ((read_side[0] == 4 and target == 1) or (read_side[0] == 2 and target == 2)):
```

Comment:
can the read_side byr array be empty ?
if yes we could be causing exception here.

Reply:
Yes, I'd better to call get_read_side() to handle all the exceptions. I will submit a commit to fix it, thanks!